src/portal/lib/src/image-name-input/image-name-input.component.ts

```diff
@@ -16,16 +16,29 @@ export class ImageNameInputComponent implements OnInit, OnDestroy {
     selectedProjectList: Project[] = [];
     proNameChecker: Subject<string> = new Subject<string>();
     imageNameForm: FormGroup;
+    public project: string;
+    public repo: string;
+    public tag: string;
 
     constructor(
         private fb: FormBuilder,
         private errorHandler: ErrorHandler,
         private proService: ProjectService,
     ) {
         this.imageNameForm = this.fb.group({
-            projectName: ["", Validators.required],
-            repoName: ["", Validators.required],
-            tagName: ["", Validators.required],
+            projectName: ["", Validators.compose([
+                Validators.minLength(2),
+                Validators.required,
+                Validators.pattern('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
+            ])],
+            repoName: ["", Validators.compose([
+                Validators.required,
+                Validators.pattern('^[a-z0-9]+(?:[._-][a-z0-9]+)*$')
```

Comment:
You can refer to the pattern in the backend for the validation, repo name also support subpath, like `aa/bb`.

https://github.com/goharbor/harbor/pull/6547/files#diff-8afa4a5b93f3fefbb482d017eaaa9f91

Reply:
done.